### ISSUE TITLE
NPM and webmin

### ISSUE THREAD

**Issue by tokar86a** [NginxProxyManager/nginx-proxy-manager#3431]
Hi

I am running Webmin behind NPM. The problem i am having is that when i am visiting Webmin admin portal i get the same loggin ip as the NPM docker contatiner host. In this case 192.168.1.118 when the ip should be 192.168.1.160. 

Any suggesting when it comes to get webmin to understand my true IP address?

I have sett the following headers in NPM advance section:

real_ip_header CF-Connecting-IP;
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

And in Webmin i have turn the following setting on: Trust remote IP address provided by proxies.

Can i somehow add another proxy_set_header ??

**Comment by tokar86a** [comment]
> i used this solution
> 
> create proxy host with classic domain webmin.domain.tld scheme https ip x.x.x.x port 10000
> 
> on webmin / virtualmin
> 
> nano /etc/webmin/config add at end referers=webmin.domain.tld
> 
> nano /etc/webmin/miniserv.conf add at end redirect_ssl=1 redirect_host=webmin.domain.tld
> 
> systemctl restart webmin
> 
> works, also file manager

I have try that thing to and i also is giving me the same IP address of 192.168.1.118 when it shold end whit 160.

**Comment by ale82x** [comment]
yes i use that config because webmin is behind NAT , so with npm i don't need to port forwarding...

but i always redirect to webmin.domain.tld

**Comment by tokar86a** [comment]
> yes i use that config because webmin is behind NAT , so with npm i don't need to port forwarding...
> 
> but i always redirect to webmin.domain.tld

So you also getting the IP of the NPM docker host then?

**Comment by ale82x** [comment]
i have webmin behind NAT, and instead port forwarding i use npm to expose wemin...

yes i check by see npm ip not real ip...

i try to use
https://github.com/webmin/webmin/issues/1027#issuecomment-1507807131

but file manager won't work...

if you have a better solution i will try

thank you

**Comment by ale82x** [comment]
i think to have real ip is port forwarding (that use layer 3 TCP, instead nginx-pm use layer 7 application HTTP)
and webmin can't "read" header X-Real-IP such as apache / nginx

**Comment by tokar86a** [comment]
> i think to have real ip is port forwarding (that use layer 3 TCP, instead nginx-pm use layer 7 application HTTP) and webmin can't "read" header X-Real-IP such as apache / nginx

The strange thing is that maybe more people should be reported about this if that was the case? Webmin staff seems to not be aware about any problem here.

**Comment by tokar86a** [comment]
Have still not been able to figure this out.

**Comment by jars101** [comment]
Hello, any updates on this?
